vite-project/src/components/mealsList.jsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import ListItemWithBorder from "./listItemWithBorder";
+import apiurl from "../apiurl";
+function MealsList() {
+  const [meals, setMeals] = useState([]);
+  const [loading, setLoading] = useState(true);
```

Comment:
this is just my personal opinion, but I think it would be better if the default loading state is false and then I would invoke it right before fetching meals, but this works as well :)